library/std/src/io/mod.rs

```diff
@@ -1155,7 +1162,9 @@ impl<'a> IoSliceMut<'a> {
         }
 
         *bufs = &mut replace(bufs, &mut [])[remove..];
-        if !bufs.is_empty() {
+        if bufs.is_empty() {
+            assert!(n == accumulated_len, "advancing io slices beyond their length");
```

Comment:
maybe use `assert_eq` here to include the expected length in the panic message?

Reply:
Then the user would just see `left: 200, right: 100`, which doesn't mean much. (Also, `assert_eq` results in mode codegen, which is a bit of a waste.)

Reply:
ok